src/hnswscan.c

```diff
@@ -185,25 +185,23 @@ hnswgettuple(IndexScanDesc scan, ScanDirection dir)
 	while (list_length(so->w) > 0)
 	{
 		HnswCandidate *hc = llast(so->w);
-		ItemPointer heaptid;
+		ItemPointerData heaptid;
 
 		/* Move to next element if no valid heap TIDs */
-		if (list_length(hc->element->heaptids) == 0)
+		if (hc->element->num_heaptids == 0)
 		{
 			so->w = list_delete_last(so->w);
 			continue;
 		}
 
-		heaptid = llast(hc->element->heaptids);
-
-		hc->element->heaptids = list_delete_last(hc->element->heaptids);
+		heaptid = HnswRemoveHeapTid(hc->element);
```

Comment:
After fetching a TID, delete it to make sure we can get the next TID.  I wonder if `HnswRemoveHeapTid` is necessary here.   How about adding a field 'curTid' in 'HnswScanOpaqueData' to help iterate the 'heaptids'?